numpy/core/_internal.py

```diff
@@ -294,8 +294,7 @@ def __init__(self, array, ptr=None):
         if ctypes:
             self._ctypes = ctypes
             # get a void pointer to the buffer, which keeps the array alive
-            self._data = _get_void_ptr(array)
-            assert self._data.value == ptr
+            self._data = self._ctypes.c_void_p(ptr)
         else:
             # fake a pointer-like object that holds onto the reference
             self._ctypes = _missing_ctypes()
```

Comment:
Does the branch below still need the `_data._objects = array` path in this case?

Reply:
The `_get_void_ptr ` is not used anymore here, I have to look at it closer to be sure if this is quite right.

Reply:
_data._objects is not needed. Here _data is not a casted pointer, it does not need to save the original pointer type.

_get_void_ptr is not needed by _ctypes class. I think it can be removed but I leave it there in case it is need by other functions.

Reply:
`_ctypes` was the only user of `_get_void_ptr` - it sounds like it can be removed too.

Reply:
Should I remove it or leave it to you if there are other places of the PR that may need to be cleaned up?

Reply:
Go ahead and remove it, along with the functions it calls